Description: Handle positional printf arguments
 This has been fixed upstream in the 2.8 branch, but after 2.8.12
Author: TIK
Bug-Debian: http://bugs.debian.org/631696
Forwarded: http://trac.wxwidgets.org/ticket/10965
Last-Update: 2012-04-25
Applied-Upstream: 2.8.13

--- a/src/common/wxchar.cpp (revision 58994)
+++ b/src/common/wxchar.cpp (revision 68568)
@@ -1500,13 +1500,42 @@
         if ( CopyFmtChar(*format++) == _T('%') )
         {
-            // skip any flags
-            while ( IsFlagChar(*format) )
-                CopyFmtChar(*format++);
-
-            // and possible width
-            if ( *format == _T('*') )
-                CopyFmtChar(*format++);
+#if wxUSE_PRINTF_POS_PARAMS
+            if( *format >= '0' && *format <= '9' )
+            {
+                SkipDigits(&format);
+                if( *format == '$' )
+                {
+                    // It was a positional printf argument
+
+                    CopyFmtChar(*format++);
+
+                    // skip any flags
+                    while ( IsFlagChar(*format) )
+                        CopyFmtChar(*format++);
+
+                    // and possible width
+                    if ( *format == _T('*') )
+                        CopyFmtChar(*format++);
+                    else
+                        SkipDigits(&format);
+                }
+                else
+                {
+                    // Must have been width
+                }
+            }
             else
-                SkipDigits(&format);
+#endif
+            {
+                // skip any flags
+                while ( IsFlagChar(*format) )
+                    CopyFmtChar(*format++);
+
+                // and possible width
+                if ( *format == _T('*') )
+                    CopyFmtChar(*format++);
+                else
+                    SkipDigits(&format);
+            }
 
             // precision?
